src/test/server.c

```diff
@@ -125,6 +137,17 @@ void *pdispatch(void *data)
         printf("In server callback for msg #%ld\n", msg->hinfo.cookie);
     }
 
+    if (!authorize(msg->hinfo.target, msg->hinfo.instance))
```

Comment:
What is the point of the instance field?  The server should look at the client cert associated with this connection, not the (untrusted) instance ID passed by the client.